core/src/banking_stage/transaction_scheduler/transaction_state.rs

```diff
@@ -34,77 +30,38 @@ pub(crate) enum TransactionState {
     /// The transaction is available for scheduling.
     Unprocessed {
         transaction_ttl: SanitizedTransactionTTL,
-        transaction_priority_details: TransactionPriorityDetails,
-        transaction_cost: TransactionCost,
-        forwarded: bool,
+        priority: u64,
+        cost: u64,
     },
     /// The transaction is currently scheduled or being processed.
-    Pending {
-        transaction_priority_details: TransactionPriorityDetails,
-        transaction_cost: TransactionCost,
-        forwarded: bool,
-    },
+    Pending { priority: u64, cost: u64 },
 }
 
 impl TransactionState {
     /// Creates a new `TransactionState` in the `Unprocessed` state.
-    pub(crate) fn new(
-        transaction_ttl: SanitizedTransactionTTL,
-        transaction_priority_details: TransactionPriorityDetails,
-        transaction_cost: TransactionCost,
-    ) -> Self {
+    pub(crate) fn new(transaction_ttl: SanitizedTransactionTTL, priority: u64, cost: u64) -> Self {
         Self::Unprocessed {
             transaction_ttl,
-            transaction_priority_details,
-            transaction_cost,
-            forwarded: false,
-        }
-    }
-
-    /// Returns a reference to the priority details of the transaction.
-    pub(crate) fn transaction_priority_details(&self) -> &TransactionPriorityDetails {
```

Comment:
for the sake of backport readability, can you read these removed functions and annotate with `#[allow(dead_code)]`?

Reply:
The meta data stored in transaction_state changed, so we no longer have a `TransactionPriorityDetails`. I don't think we should add back those unused metadata and code for calculating them.

I could add a block-comment around these removed fns if you think that'd help simplify the diff.

Reply:
Added block-comment from 50 to 97: c80d5fb2e9fcae399e451894dfcda470f8a1c0dd